frontend/src/components/TreeView.js

```diff
@@ -0,0 +1,207 @@
+import React, { useEffect, useState } from 'react'
+import { useNavigate, useLocation } from 'react-router-dom'
+import { Text, Box } from '@chakra-ui/react'
+const TreeView = (props) => {
+    const navigate = useNavigate()
+    const location = useLocation()
+    const { tickets } = props
+    const [tree, setTree] = useState([])
+    const pathname = location.pathname.substring(1).split('/')
+    const ticketType = pathname[0]
+    const ticketID = parseInt(pathname[1])
+
+    const sortAndTypeTickets = (list, type) => {
+        let newArrayWithType = list.map((obj) => ({ ...obj, type: type }))
+        let sortedArr = newArrayWithType.sort((a, b) =>
+            a.name > b.name ? 1 : -1
+        )
+        return sortedArr
+    }
+
+    let { SF, FI, PPR, UPR } = tickets
+
+    if (SF.length !== 0) {
+        SF = sortAndTypeTickets(SF, 'SF')
+        FI = sortAndTypeTickets(FI, 'FI')
+        PPR = sortAndTypeTickets(PPR, 'PPR')
+        UPR = sortAndTypeTickets(UPR, 'UPR')
+    }
+
+    useEffect(() => {
+        if (SF.length !== 0) {
+            let id = ticketID
+            if (ticketType === 'FI') {
+                //use SF_link to get SF
+                let result = FI.find((ticket) => ticket._id === ticketID)
+                id = result.sf_link
+            } else if (ticketType === 'PPR') {
+                //trace link to FI and then to SF
+                let result = PPR.find((ticket) => ticket._id === ticketID)
+                const FI_ID = result.fi_link
+                result = FI.find((ticket) => ticket._id === FI_ID)
+                id = result.sf_link
+            } else if (ticketType === 'UPR') {
+                let result = UPR.find((ticket) => ticket._id === ticketID)
+                const FI_ID = result.fi_link
+                result = FI.find((ticket) => ticket._id === FI_ID)
+                id = result.sf_link
+            }
+            setTreeFromSF(id)
+            //search FIs to find if they have ticketID, return a list of FIs
+            //for each FIs, search PPR and UPR if they have have that FI, return list of PPR and a list of UPR
+        }
+    }, [tickets, location])
+
+    const setTreeFromSF = (sf_id) => {
```

Comment:
This entire function is done via the /sponsorshipfunds/getallchildren/:id endpoint, as long as we have the sf_id we can get all the data by calling that endpoint as opposed to rewriting the logic